integration_tests/features/CoinbaseReorg.feature

```diff
@@ -0,0 +1,78 @@
+    Then wallet WALLET_A1 has 10 coinbase transactions
+#    Then wallet WALLET_A1 detects at least 7 coinbase transactions as Mined_Confirmed
+        # Use 7 of the 10 coinbase UTXOs in transactions (others require 3 confirmations)
+    And I send 1000000 tari from wallet WALLET_A1 to wallet WALLET_A2 at fee 100
+    And I send 1000000 tari from wallet WALLET_A1 to wallet WALLET_A2 at fee 100
+    And I send 1000000 tari from wallet WALLET_A1 to wallet WALLET_A2 at fee 100
+    And I send 1000000 tari from wallet WALLET_A1 to wallet WALLET_A2 at fee 100
+    And I send 1000000 tari from wallet WALLET_A1 to wallet WALLET_A2 at fee 100
+    And I send 1000000 tari from wallet WALLET_A1 to wallet WALLET_A2 at fee 100
+    And I send 1000000 tari from wallet WALLET_A1 to wallet WALLET_A2 at fee 100
+    Then wallet WALLET_A1 detects all transactions are at least Broadcast
+        #
+        # Chain 2:
+        #   Collect 5 coinbases into one wallet, send Txn, advance to height 10
+        #
+    And I have a seed node SEED_B
+        # Add multiple base nodes to ensure more robust comms
+    And I have a base node NODE_B1 connected to seed SEED_B
+    And I have a base node NODE_B2 connected to seed SEED_B
+    And I have a base node NODE_B3 connected to seed SEED_B
+    And I have wallet WALLET_B1 connected to seed node SEED_B
+    And I have wallet WALLET_B2 connected to seed node SEED_B
+    And I have a merge mining proxy PROXY_B connected to SEED_B and WALLET_B1
+    When I merge mine 10 blocks via PROXY_B
+    Then all nodes are at height 10
+        # Wait for coinbase statuses to change in the wallet
+    #When I wait 60 seconds
+    And I list all coinbase transactions for wallet WALLET_B1
+    Then wallet WALLET_B1 has 10 coinbase transactions
+#    Then wallet WALLET_B1 detects at least 7 coinbase transactions as Mined_Confirmed
+        # Use 7 of the 10 coinbase UTXOs in transactions (others require 3 confirmations)
+    And I send 1000000 tari from wallet WALLET_B1 to wallet WALLET_B2 at fee 100
+    And I send 1000000 tari from wallet WALLET_B1 to wallet WALLET_B2 at fee 100
+    And I send 1000000 tari from wallet WALLET_B1 to wallet WALLET_B2 at fee 100
+    And I send 1000000 tari from wallet WALLET_B1 to wallet WALLET_B2 at fee 100
+    And I send 1000000 tari from wallet WALLET_B1 to wallet WALLET_B2 at fee 100
+    And I send 1000000 tari from wallet WALLET_B1 to wallet WALLET_B2 at fee 100
+    And I send 1000000 tari from wallet WALLET_B1 to wallet WALLET_B2 at fee 100
+    Then wallet WALLET_B1 detects all transactions are at least Broadcast
+        #
+        # Connect Chain 1 and 2
+        #
+        # TODO: This wait is needed to stop next base node task from continuing
+    When I wait 1 seconds
+    And I have a base node NODE_C connected to all seed nodes
+        # Wait for the reorg to filter through
+    When I wait 30 seconds
+    Then all nodes are at height 10
```

Comment:
I think rather mine a significantly stronger chain on B, so that it always reorgs from A->B. It should make it easier to test specific values and statuses on A and B in the following steps. We can't guarantee 15 blocks will be stronger than 10 at the moment, but we might be able to in the future. Until then, if it occasionally fails because it reorgs to A instead, that isn't a train smash

Reply:
I do not know if I agree with this approach, though. The current test metrics do not care which chain wins, just that the invalid coinbase transactions will be discarded when the tip moves ahead. The outcome is always predictable, and when the test fails something is definitely wrong.